src/domain/challenge/opportunity/opportunity.lifecycle.options.provider.ts

```diff
@@ -0,0 +1,62 @@
+import { Inject, Injectable, LoggerService } from '@nestjs/common';
+import { LogContext } from '@common/enums';
+import { WINSTON_MODULE_NEST_PROVIDER } from 'nest-winston';
+import { MachineOptions } from 'xstate';
+import { LifecycleService } from '@domain/common/lifecycle/lifecycle.service';
+import { OpportunityService } from './opportunity.service';
+import {
+  IOpportunity,
+  OpportunityLifecycleEventInput,
+} from '@domain/challenge/opportunity';
+
+@Injectable()
+export class OpportunityLifecycleOptionsProvider {
+  constructor(
+    private lifecycleService: LifecycleService,
+    private opportunityService: OpportunityService,
+    @Inject(WINSTON_MODULE_NEST_PROVIDER) private readonly logger: LoggerService
+  ) {}
+
+  async eventOnOpportunity(
+    opportunityEventData: OpportunityLifecycleEventInput
```

Comment:
For discussion - I feel this (and again as a pattern) should be OpportunityEventInput. The event is on opportunity and is being used by the lifecycle, but logically it's an opportunity event, not opportunityLifecycle event. The lifecycle is essentially a stateMachine consuming events of the underlying entities. Moving that makes explicit that that event is also on the level of the entity and not the lifecycle (in my opinion).

Reply:
Agreed, ties in with the actual mutation name then.